third_party/amd/lib/TritonAMDGPUTransforms/RefineOps.cpp

```diff
@@ -668,12 +671,14 @@ struct TritonAMDGPURefineOps
         }
       });
 
+      int32_t dotIndex = 0;
       block->walk([&](triton::DotOp dotOp) {
         OpBuilder rewriter(dotOp->getContext());
         // TODO: extend to WMMA instructions
-        if (failed(rewriteMFMA(rewriter, dotOp))) {
+        if (failed(rewriteMFMA(rewriter, dotOp, dotIndex))) {
           LDBG("failed to refine tt.dotOp: " << *dotOp);
-        }
+        } else
```

Comment:
We may want to increment the dotIndex even when we fail, so keep track of there was another dot even though we couldn't refine it.